### ISSUE TITLE
Training YOLOv3 on COCO dataset

### ISSUE THREAD

**Issue by smohan10** [wizyoung/YOLOv3_TensorFlow#84]
Hi, 
I am trying to use your code to run training on MSCOCO dataset. I trained for 200 epochs for over a week and the max mAP I am getting is 0.45 for 416x416 resolution image. This is lower compared to what is quoted on paper which is about 0.55. 

Attached the tensorboard result. 

![yolov3](https://user-images.githubusercontent.com/5778892/58648775-4fc22a00-82bf-11e9-8108-5666bfe1554f.PNG)

Has anyone tried training on COCO dataset?

**Comment by tianfengyijiu** [comment]
I have trained on PASCAL VOC, and I did not get a good result, just 62% mAP, not 75.9%

**Comment by smohan10** [comment]
Just checking again if anyone has tried training on COCO dataset.

**Comment by ZH-Lee** [comment]
> I have trained on PASCAL VOC, and I did not get a good result, just 62% mAP, not 75.9%

I have trained on VOC2007, but i got mAp 0, recall 0 after 10 epochs. is it happened during your training?

**Comment by wizyoung** [comment]
Thanks for all your test. It seems there are some bugs in my loss function which can be found in issue #85 ( issue reported in Chinese). Currently I have no time to fix that. If any of you have time, you can correct that mistake which is not difficult

**Comment by wizyoung** [comment]
@tianfengyijiu 62% mAP is too low by the way. I have trained the mAP on VOC to 70% before using the current buggy code

**Comment by ZH-Lee** [comment]
> Thanks for all your test. It seems there are some bugs in my loss function which can be found in issue #85 ( issue reported in Chinese). Currently I have no time to fix that. If any of you have time, you can correct that mistake which is not difficult

请问，为什么我用VOC去训练，得到的mAP和召回为0呢？我的class名字也替换成了voc的类名，其他基本没变过。

**Comment by ZH-Lee** [comment]
> @tianfengyijiu 62% mAP is too low by the way. I have trained the mAP on VOC to 70% before using the current buggy code

loss一直在降低，但是检测出来的bbox都是no boxes

**Comment by tianfengyijiu** [comment]
> > @tianfengyijiu 62% mAP is too low by the way. I have trained the mAP on VOC to 70% before using the current buggy code
> 
> loss一直在降低，但是检测出来的bbox都是no boxes

数据按格式生成对了吗

**Comment by ZH-Lee** [comment]
> > > @tianfengyijiu 62% mAP is too low by the way. I have trained the mAP on VOC to 70% before using the current buggy code
> > 
> > 
> > loss一直在降低，但是检测出来的bbox都是no boxes
> 
> 数据按格式生成对了吗

对了的，不然loss也不会降低嘛。

**Comment by tianfengyijiu** [comment]
> > > > @tianfengyijiu 62% mAP is too low by the way. I have trained the mAP on VOC to 70% before using the current buggy code
> > > 
> > > 
> > > loss一直在降低，但是检测出来的bbox都是no boxes
> > 
> > 
> > 数据按格式生成对了吗
> 
> 对了的，不然loss也不会降低嘛。

贴个evaluate的结果啊

**Comment by tianfengyijiu** [comment]
> @tianfengyijiu 62% mAP is too low by the way. I have trained the mAP on VOC to 70% before using the current buggy code

我加了一些随机数据增强才到70以上，另外麻烦问下你测试的时候的几个阈值是多少。

**Comment by factplay1** [comment]
Anyone had any luck on this?
I can't go beyond mAP of 8% when I train CoCo from scratch, even after 2 weeks of training!

**Comment by factplay1** [comment]
@smohan10 : Are you training entirely from scratch over the whole model, or only learning the Yolo-head part (with darknet body loaded from a pretrained model)?

**Comment by wizyoung** [comment]
@smohan10 I think I've fixed the bugs. By the way you can set `score_threshold` to a lower value like 0.01 or 0.005 for higher mAP

**Comment by wizyoung** [comment]
@tianfengyijiu I made a quick test on VOC dataset by only training the yolo-head part for 20 epochs. And the mAP on the test set is now 73.4%. The result should be better if I finetune the whole model.

**Comment by wizyoung** [comment]
I made two quick tests on VOC dataset. (1) loading darknet weights I got ~77 mAP with 416*416 input; (2) loading weighs from all layers except last three conv2d layers in 3 scales, I got ~84 mAP with 416*416 input. No try-hard fine-tuning. Performance seems normal.

**Comment by wizyoung** [comment]
Now it should be fixed.
